src/Symfony/Component/HttpKernel/Event/GetResponseForExceptionEvent.php

```diff
@@ -44,6 +40,20 @@ public function __construct(HttpKernelInterface $kernel, Request $request, int $
      * @return \Exception The thrown exception
      */
     public function getException()
+    {
+        if (!$this->exception instanceof \Exception) {
+            return new ErrorException($this->exception);
+        }
+
+        return $this->exception;
+    }
+
+    /**
+     * @internal to be removed in Symfony 5.0 when getException() can return a \Throwable instead of an \Exception
```

Comment:
I don't the upgrade path here: on 5.0, `getException()` will suddenly return a Throwable, without any previous notice in 4.4. Either we're good with this and we can do it on 4.4 already, or we need another path since we don't do hard BC breaks, isn't it?

This maps to my other comment on catching throwables in handle().
If we don't catch throwable there, we can update this to return a throwable - because in practice we will never use it with an `Error`.

Reply:
For @Tobion: returning a \Throwable in getException() on master directly would be a BC break. This is why all changes were reverted in this event class and Errors are transformed to ErrorException before instantiating it.

Reply:
What's the point of this all these changes if ExceptionEvent cannot accept Throwable?

Reply:
> \Throwable in getException() on master directly would be a BC break

That is why we would do it 5.0 and not in 4.4.